src/rpg_database/character_sheet/Race.java

```diff
@@ -2,18 +2,56 @@
 
 import static rpg_database.character_sheet.common.CharacterSheetCommon.generateEnumText;
 
-public enum Race {
+import java.util.HashSet;
 
-	HUMAN, ELF, DWARF, QUNARI;
+import rpg_database.character_sheet.exceptions.CharacterSheetException;
+import rpg_database.character_sheet.interfaces.CustomSetter;
+
+public enum Race implements CustomSetter<Race> {
+
+	HUMAN(10), ELF(12), DWARF(8), QUNARI(10);
 
 	private final String text;
+	private final int speed;
 
-	private Race() {
+	private Race(int speed) {
 		this.text = generateEnumText(this.name());
+		this.speed = speed;
+	}
+
+	public int getBaseSpeed() {
+		return this.speed;
 	}
 
 	@Override
 	public String toString() {
 		return text;
 	}
+
+	@Override
+	public Class<Race> getImplementingClass() {
+		return Race.class;
+	}
+
+	@Override
+	public void setSelfInSheet(CharacterSheet characterSheet) {
+		HashSet<Race> allowedRaces = getAllowedRacesForBackground(getActaulBackground(characterSheet));
+		if (allowedRaces.size() > 1)
+			if (allowedRaces.contains(this))
+				characterSheet.characterData.put(Fields.RACE, this);
+			else
+				throw new CharacterSheetException(String.format("%s does not offer %s race for choicing!", getActaulBackground(characterSheet)
+						.toString(), this.text));
+		else
+			throw new CharacterSheetException("The race can not be set manually!");
+	}
+
+	private HashSet<Race> getAllowedRacesForBackground(Background background) {
+		return background.getAllowedRaces();
+	}
+
+	private Background getActaulBackground(CharacterSheet characterSheet) {

```

Comment:
Én ennek azt a nevet adtam volna, hogy getCharacterSheetBackground, amúgy az Actual-t elírtad. 
Szerintem lehet egyszerűbb lenne ezt pl. egy függvény helyett egy változóval megoldani, mert értékadáson kívül nem történik különösebb dolog, viszont 2x is hívod ahelyett hogy egy mentett objektummal dolgoznál: itt. pl egy Background characterSheetBackground = blablabla-ba menteni, amit aztán mindenhol tudsz használni

Reply:
igen, eredetileg azt volt, csak gondoltam a setSelfInSheet-et ne árasszuk el átláthatatlan kóddal